src/Application.vala

```diff
@@ -8,6 +8,25 @@ public class Dock.Application : Gtk.Application {
         Object (application_id: "io.elementary.dock");
     }
 
+    protected override void startup () {
+        base.startup ();
+
+        var granite_settings = Granite.Settings.get_default ();
+        var gtk_settings = Gtk.Settings.get_default ();
+
+        gtk_settings.gtk_icon_theme_name = "elementary";
```

Comment:
I don't think we want to force the icon theme here, since we are dealing with application icons, and the user can be using a custom theme that override some application icons.